### PR TITLE
factories no longer depends on initializers

### PR DESCRIPTION
**Pull request by svycka**

### CODE REVIEW

File: src/Service/Factory/AdapterManagerFactory.php [hunk 1]
```diff
@@ -80,6 +80,9 @@ public function createService(ServiceLocatorInterface $serviceLocator)
 
         $serviceConfig = new Config($serviceConfig);
 
-        return new AdapterManager($serviceConfig);
+        $adapterManager = new AdapterManager($serviceConfig);
+        $adapterManager->setServiceLocator($serviceLocator);
```
Comment (svycka) [inline src/Service/Factory/AdapterManagerFactory.php]: `$serviceLocator` should be set in constructor, but if we need to support php 5.4 I have to set it like this :D

**Comment by svycka** [comment]
creating this PR I have noticed that this module has some problems with lowest dependencies for example you use in tests `$serviceManager->isShared()` method added in 2.4 version if I remember correctly but in your composer you have ~2.2 version also phpunit version should be updated and few more problems. if you want to fix them you can try by instaling packages like this `composer update --prefer-lowest`

I did not have time to create SMv3 supported version(maybe when I will have time), but this one at least does not throw deprecated notices.

**Comment by basz** [comment]
https problems for pear too... I'll try if I can find some time to fix a few problems. btw I have nothing against dropping php 5.4 (do you depend on that). But I think I shouldn't do that in a patch release to avoid problems for anyone using this in production. Thank you for the PR (i'll do that asap) and tips!

File: config/module.config.php [hunk 2]
```diff
@@ -38,11 +38,12 @@
         ],
     ],
     'service_manager' => [
-        'invokables' => [
-            'BsbFlysystemManager' => 'BsbFlysystem\Service\FilesystemManager',
+        'aliases' => [
+            'BsbFlysystem\Service\FilesystemManager' => 'BsbFlysystemManager',
```
Comment (svycka) [inline config/module.config.php]: this maybe not really required as it is not documented, but many people use `BsbFlysystem\Service\FilesystemManager::class` syntax

Comment (basz) [inline config/module.config.php]: I prefer that too, but that's not supported in PHP5.4 Definitely will be using that once 5.4 support is dropped

Comment (svycka) [inline config/module.config.php]: @basz the change allows them to use this syntax now if they are on PHP>=5.5

**Comment by svycka** [comment]
Added tests to check if `BsbFlysystem\Service\FilesystemManager::class` really works :D should be ok now

**Comment by basz** [comment]
To run the travis tests I think you should the following to the composer.json until PEAR works again. But they seem to not be fixed, reading seldeak's comments: https://github.com/composer/composer/issues/5069


```
    ,
    "config" : {
        "secure-http" : false
    },
```

**Comment by svycka** [comment]
> btw I have nothing against dropping php 5.4 (do you depend on that)

lets do release with these changes for people with PHP5.4. Then release new version with SMv3 support where we could change service manager to minimum 2.7.5 version.

>     ,
    "config" : {
        "secure-http" : false
    },

should I add this to test this on travis? because locally I have no issues.

**Comment by basz** [comment]
Do you have that option set globally then? Or an outdated composer.phar? I couldn't install the dependancies locally until I specifically turned of that option.

**Comment by svycka** [comment]
@basz maybe works is wrong word :D

```
$ composer update
Loading composer repositories with package information
Initializing PEAR repository http://pear.php.net
PEAR repository from http://pear.php.net could not be loaded. Your configuration does not allow connections to http://pear.php.net/channel.xml. See https://getcomposer.org/doc/06-config.md#secure-http for details.
Updating dependencies (including require-dev)
...
```
but it installs everything else and tests pass so I call that works :)

So should I add the `"secure-http" : false` option? maybe you can do this in another PR I will rebase this if you want

**Comment by basz** [comment]
I'm on it... patch to rebase on pending

**Comment by basz** [comment]
please rebase on master and repush so travis is happy

**Comment by svycka** [comment]
done

**Comment by basz** [comment]
thank you! a new release has been published

**Pull request merged**